src/test/java/seedu/address/model/employee/ContainsAllPredicateTest.java

```diff
@@ -1,75 +1,181 @@
+        ContainsAllPredicate predicate = createContainsAllPredicateHelper("name", keyword);
+        Employee employee = new EmployeeBuilder().withName(keyword).build();
+        assertTrue(predicate.test(employee));
+    }
+
+    @Test
+    public void test_doesNotContainName_returnsFalse() {
+        String keyword = VALID_NAME_AMY;
+        ContainsAllPredicate predicate = createContainsAllPredicateHelper("name", keyword);
+        Employee employee = new EmployeeBuilder().withName(VALID_NAME_BOB).build();
+        assertFalse(predicate.test(employee));
+    }
+
+    @Test
+    public void test_containsSalary_returnsTrue() {
+        String keyword = VALID_SALARY_AMY;
+        ContainsAllPredicate predicate = createContainsAllPredicateHelper("salary", keyword);
+        Employee employee = new EmployeeBuilder().withSalary(keyword).build();
+        assertTrue(predicate.test(employee));
+    }
+
     @Test
     public void test_containsAddress_returnsTrue() {
-        // One keyword
         String keyword = VALID_ADDRESS_BOB;
-        Set<Address> addressSet = new CustomSet<>();
-        addressSet.add(new Address(keyword));
-        ContainsAllPredicate predicate = new ContainsAllPredicate(new CustomSet<>(), new CustomSet<>(),
-                new CustomSet<>(), addressSet, new CustomSet<>(), new CustomSet<>(), new CustomSet<>(),
-                new CustomSet<>(), new CustomSet<>());
-        System.out.println(predicate);
-        System.out.println(new EmployeeBuilder().withAddress(keyword).build());
-        assertTrue(predicate.test(new EmployeeBuilder().withAddress(keyword).build()));
+        ContainsAllPredicate predicate = createContainsAllPredicateHelper("address", keyword);
+        Employee employee = new EmployeeBuilder().withAddress(keyword).build();
+        assertTrue(predicate.test(employee));
     }
 
+    @Test
+    public void test_doesNotContainsAddress_returnsFalse() {
+        String keyword = VALID_ADDRESS_BOB;
+        ContainsAllPredicate predicate = createContainsAllPredicateHelper("address", keyword);
+        Employee employee = new EmployeeBuilder().withAddress(VALID_ADDRESS_AMY).build();
+        assertFalse(predicate.test(employee));
+    }
 
     @Test
     public void test_emptyPredicate_returnsTrue() {
-        // Zero keywords
-        ContainsAllPredicate predicate = new ContainsAllPredicate(new CustomSet<>(), new CustomSet<>(),
-                new CustomSet<>(), new CustomSet<>(), new CustomSet<>(), new CustomSet<>(), new CustomSet<>(),
-                new CustomSet<>(), new CustomSet<>());
-        assertTrue(predicate.test(new EmployeeBuilder().withDepartments("R&D").build()));
+        ContainsAllPredicate predicate = createContainsAllPredicateHelper("", "");
+        Employee employee = new EmployeeBuilder().build();
+        assertTrue(predicate.test(employee));
     }
 
     @Test
-    public void test_differentAttributeValueBetweenPredicateAndEmployee_returnsFalse() {
```

Comment:
This test too -- why was it removed?

Reply:
replaced by test_doesNotContainsAddress (line 113)

it tests the same thing, between amy's and bob's address -> expected to return false